village/src/components/Smurf.js

```diff
@@ -1,13 +1,51 @@
 import React from 'react';
+import EditSmurf from './EditSmurf'
 
-const Smurf = props => {
-  return (
-    <div className="Smurf">
-      <h3>{props.name}</h3>
-      <strong>{props.height} tall</strong>
-      <p>{props.age} smurf years old</p>
-    </div>
-  );
+class Smurf extends React.Component{
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      edit: false
+    }
+  }
+
+  edit = (edit) => {
+    let id = this.props.id
+    this.props.edit(id, edit)
+    this.setState({edit: false});
+  }
+
+  clickHandler = () => {
+    if (this.state.edit) {
+      this.setState({edit: false})
+    } else{
+      this.setState({edit: true})
+    }
+  }
+
+  Form = () => {
+    if (this.state.edit) {
+      return <EditSmurf edit={this.edit}/>
+    } else {
+      return null;
+    }
+  }
+
+  render() {
+    return (
+      <div className="Smurf">
+        <h3>{this.props.name}</h3>
+        <strong>{this.props.height} tall</strong>
+        <p>{this.props.age} smurf years old</p>
+        <div>
+          <span onClick={ () => this.props.deleteSmurf(this.props.id)}>Delete</span>
+          <span onClick={ () => this.clickHandler()} >Edit</span>
+        </div>
+        <this.Form />
```

Comment:
You could just call this like 
```suggestion
        {this.state.edit && <EditSmurf edit={this.edit}/>}
```
So what that is saying, since `<EditSmurf edit={this.edit}/>` will always be `true` is if `this.state.edit` is true, render `EditSmurf`